ci/fireci/fireci/emulator.py

```diff
@@ -12,19 +12,127 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import datetime
 import logging
+import os
+import signal
+import subprocess
+import time
 
 _logger = logging.getLogger('fireci.emulator')
 
+EMULATOR_BINARY = 'emulator'
+ADB_BINARY = 'adb'
+EMULATOR_NAME = 'test'
+
+EMULATOR_FLAGS = ['-no-audio', '-no-window', '-skin', '768x1280']
+
 
-# TODO(vkryachko): start/shutdown the emulator.
 class EmulatorHandler:
+  """
+     Context manager that launches an android emulator for the duration of its execution.
 
-  def __init__(self, artifacts_dir):
+     As part of its run it:
+     * Launches the emulator
+     * Waits for it to boot
+     * Starts logcat to store on-device logs
+     * Produces stdout.log, stderr.log, logcat.log in the artifacts directory
+  """
+
+  def __init__(
+      self,
+      artifacts_dir,
+      *,
+      name=EMULATOR_NAME,
+      emulator_binary=EMULATOR_BINARY,
+      adb_binary=ADB_BINARY,
+      # for testing only
+      emulator_stdin=None,
+      wait_for_device_stdin=None,
+      logcat_stdin=None):
     self._artifacts_dir = artifacts_dir
 
+    log_dir = '{}_emulator'.format(name)
+    self._stdout = self._open(log_dir, 'stdout.log')
+    self._stderr = self._open(log_dir, 'stderr.log')
+    self._adb_log = self._open(log_dir, 'logcat.log')
+    self._name = name
+
+    self._emulator_binary = emulator_binary
+    self._adb_binary = adb_binary
+
+    self._emulator_stdin = emulator_stdin
+    self._wait_for_device_stdin = wait_for_device_stdin
+    self._logcat_stdin = logcat_stdin
+
   def __enter__(self):
-    _logger.debug('Pretend to start the emulator(TODO)')
+    _logger.info('Starting avd "{}..."'.format(self._name))
```

Comment:
nit: the double quotes should probably not encompass ellipsis.  Currently, this will look like this:
Starting avd "test...", while we'd probably want Starting avd "test"...